## **1. Ubah Nilai CF gejala dan amati perubahan hasil**

In [1]:
gejala_user = {
    "demam": 0.7,
    "batuk": 0.5,
    "sakit_tenggorokan": 0.6
}

pengetahuan = {
    "flu": {
        "demam": 0.8,
        "batuk": 0.7,
        "sakit_tenggorokan": 0.6
    }
}

def hitung_cf(gejala_user, pengetahuan_pakar):
  cf_total = 0
  first = True
  for gejala, cf_user in gejala_user.items():
    if gejala in pengetahuan_pakar:
      cf_pakar = pengetahuan_pakar[gejala]
      cf = cf_user * cf_pakar
      if first:
          cf_total = cf
          first = False
      else:
          cf_total = cf_total + cf * (1 - cf_total)
  return cf_total

cf_flu = hitung_cf(gejala_user, pengetahuan["flu"])
print(f"CF diagnosis Flu: {cf_flu:.2f}")

CF diagnosis Flu: 0.82


## **2. Tambahkan 5 gejala baru beserta nilai CF**

In [5]:
# Tambahkan nilai gejala baru ke gejala_user dan pengetahuan
# Nilai CF gejala dari user
gejala_user = {
    "demam": 0.2,  # Diubah dari 0.7 menjadi 0.2 -> pengaruh gejala demam terhadap diagnosa jadi lebih kecil
    "batuk": 0.5,
    "sakit_tenggorokan": 0.6,
    "pilek": 0.8,
    "sakit_kepala": 0.6,
    "kelelahan": 0.5,
    "nyeri_otot": 0.4,
    "hidung_tersumbat": 0.7
}

# Nilai CF dari pakar (aturan penyakit flu)
pengetahuan = {
    "flu": {
        "demam": 0.8,
        "batuk": 0.7,
        "sakit_tenggorokan": 0.6,
        "pilek": 0.9,
        "sakit_kepala": 0.5,
        "kelelahan": 0.6,
        "nyeri_otot": 0.5,
        "hidung_tersumbat": 0.7
    }
}

# Fungsi untuk menghitung nilai Certainty Factor total
def hitung_cf(gejala_user, pengetahuan_pakar):
    cf_total = 0
    first = True
    for gejala, cf_user in gejala_user.items():
        if gejala in pengetahuan_pakar:
            cf_pakar = pengetahuan_pakar[gejala]
            cf = cf_user * cf_pakar  # Mengalikan nilai kepercayaan user dan pakar
            if first:
                cf_total = cf  # Nilai awal
                first = False
            else:
                # Kombinasi CF: cf1 + cf2 * (1 - cf1)
                cf_total = cf_total + cf * (1 - cf_total)
    return cf_total

# Hitung CF diagnosis flu
cf_flu = hitung_cf(gejala_user, pengetahuan["flu"])
print(f"CF diagnosis Flu: {cf_flu:.2f}")

CF diagnosis Flu: 0.98


## **3. Jika "demam" diubah jadi 0.2, bagaimana hasilnya?**

Penjelasan:
Penurunan nilai "demam" ke 0.2 hanya berdampak kecil karena gejala-gejala lain memiliki nilai tinggi dan mendukung kuat diagnosis flu.
CF diagnosis flu tetap tinggi, yaitu sekitar 0.98.

## **4. Jelaskan bagaimana hasil perubahannya dalam bentuk komentar atau teks cell**

Penjelasan:


*   Nilai CF diagnosis Flu masih sangat tinggi (≈ 0.98) meskipun nilai gejala "demam" diturunkan dari 0.7 ke 0.2.
*   Hal ini disebabkan gejala-gejala lain (seperti pilek, batuk, dan hidung tersumbat) masih memiliki kontribusi besar terhadap total CF.
*   Karena metode kombinasi CF bersifat akumulatif, satu gejala dengan nilai rendah tidak cukup untuk menurunkan nilai akhir secara drastis.
